src/it/java/io/fabric8/elasticsearch/ElasticsearchIntegrationTest.java

```diff
@@ -304,16 +299,29 @@ protected void givenDocumentIsIndexed(String index, String type, String id, XCon
             .get();
     }
     
-    protected void givenUserIsClusterAdmin(String user) {
-        expSubjectAcccessReviewToBe(Boolean.TRUE, user);
+    protected void givenUserIsClusterAdmin(String user, String token) {
+        expSubjectAcccessReviewToBe(Boolean.TRUE, user, token);
     }
 
-    protected void givenUserIsNotClusterAdmin(String user) {
-        expSubjectAcccessReviewToBe(Boolean.FALSE, user);
+    protected void givenUserIsNotClusterAdmin(String user, String token) {
+        expSubjectAcccessReviewToBe(Boolean.FALSE, user, token);
     }
     
-    private void expSubjectAcccessReviewToBe(boolean value, String user) {
+    protected void givenTokenIsAuthorizedForUser(String username) {
+        if(username.contains("\\")){
+            username = username.replace("\\", "\\\\");
+        }
+        String response = "{\"metadata\":{\"name\":\"" + username + "\"}}";
+        apiServer.expect().get().withPath("/oapi/v1/users/~").andReturn(200,response).once();
+    }
+
+    protected void givenTokenIsUnAuthorizedForUser() {
+        apiServer.expect().get().withPath("/oapi/v1/users/~").andReturn(401,"Unauthorized").once();
+    }
+    
+    private void expSubjectAcccessReviewToBe(boolean value, String user, String token) {
         testContext.put(USERNAME, user);
+        testContext.put(TOKEN, token);
```

Comment:
Where is `testContext` used here?

Reply:
someplace to stuff values for reuse in subsequent steps.  Its really to make the test read nicer without the need to return values from the statments